sdk/core/azure_core_amqp/Cargo.toml

```diff
@@ -25,19 +25,22 @@ fe2o3-amqp-cbs = { workspace = true, optional = true }
 fe2o3-amqp-ext = { workspace = true, optional = true }
 fe2o3-amqp-management = { workspace = true, optional = true }
 fe2o3-amqp-types = { workspace = true, optional = true }
+serde.workspace = true
```

Comment:
Did you open a bug for me to consider making this optional so that our macros don't require it? Alternatively, I could add an overload to the macro so that you can do this without making it an optional feature. Actually, now that I say that out loud, we'd have to anyway because any other project that pulls in `serde` would enable it and you'd be hosed.

Reply:
No I didn't - I looked at tweaking the dependencies or adding something appropriate to the config and decided it wasn't  worth the effort to avoid referencing a package that isn't going to be used.

Reply:
It actually wouldn't be hard to opt out of `serde`. Look at how I have an "internal" override that I use to do common things: <https://github.com/Azure/azure-sdk-for-rust/blob/845eceedaf719a49be8a0b52457c5887af182c1d/sdk/typespec/typespec_client_core/src/macros.rs#L87>. I could have a similar override like `create_extensible_enum!(no_serde, ...)`.

Reply:
As discussed offline, it's not worth it since you depend on `azure_core` which pulls in `serde` invariably, so it's a sunk cost.